index.d.ts

```diff
@@ -0,0 +1,202 @@
+import * as http from 'http';
+
+export = Router;
+
+declare namespace Router {
+
+  interface RouterOptions {
+    strict?: boolean;
+    caseSensitive?: boolean;
+    mergeParams?: boolean;
+  }
+
+  interface Layer {
+    name: string,
+    handle: RequestHandler | ErrorRequestHandler,
+    handle_request: RequestHandler,
+    handle_error: ErrorRequestHandler,
+    match: (path: string) => boolean,
+  }
+
+  interface IncomingRequest extends http.IncomingMessage {
+    url: string,
```

Comment:
Isn't the url property already a member of http.IncomingMessage ? (please forgive me on some of these typescript questions -- they may be obvious to someone who knows typescript).

Reply:
Nope. That doesn't have to be there. Removed.

I'm not an expert on TS either - it would be nice to have someone, who fits the description, go over it though.